pkg/aws/cloud.go

```diff
@@ -0,0 +1,69 @@
+package aws
+
+import (
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/aws/metrics"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/aws/services"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/aws/throttle"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type Cloud interface {
```

Comment:
The name `Cloud` is my least favorite part of v1. It's basically a wrapper/facade around the SDK right?

Reply:
yep, do you have a better name?

Reply:
AppMesh and Cloud map are both AWS specific, so how about just `AWSService`?